### DISCUSSION TITLE
wanted to understand how to deploy graphql on AWS EC2 instances using graphql mesh and hive gateway

### DISCUSSION THREAD

**Discussion by vibhor-d11** [graphql-hive/gateway#123] [Q&A] [answered]
I have few doubts regarding using graphql mesh and deploying is using hive gateway without any subcription or cloud account of hive.
1. I can create mesh config and can generate the graphql file using mesh. Will also create gateway config for caching etc.
2. I can start the server using `npx hive-gateway supergraph xyz.graphql`, now if i write a script to run this server on a EC2 instance, it will serve as a graphql server right? Or do i need hive token etc to deploy it etc?
3. Also their is Dashboard section with explorer, laboratory, are those features only for hive organisation membership? or it can be used without any membership and how? could not find any documentation or same.
4. What is the best way to deploy it for best performance? using npx command and js package or as a express server or using binary cli etc?

**Comment by ardatan** [comment]
1. Yes that's how you should do!
2. You can start the server exactly like that. You need Hive CDN Token if you use Hive CDN. If you need to report to Hive Dashboard, you need Hive Token. Other than, you don't need any token if you just want to serve your API.
- https://the-guild.dev/graphql/hive/docs/high-availability-cdn
- https://the-guild.dev/graphql/hive/docs/schema-registry/usage-reporting
3. You can also self-host Hive Dashboard on your own too, if you don't want to sign up Hive.
https://the-guild.dev/graphql/hive/docs/self-hosting/get-started
4. For the best performance, you can use Hive Gateway CLI or our official Docker image. We recommend you to avoid any complicated server setup with Express etc.

**Comment by enisdenjo** [comment, marked as answer]
Hey there! Thanks for writing. Let me jump straight in:

1. GraphQL Mesh outputs a supergraph SDL that the Hive Gateway consumes. It does not generate a config file for the gateway.
2. Hive Console (the dashboard) and Hive Gateway are separate products and can be used fully independently. So yes, you can serve any Federation supergraph without ever depending, or even having an account, on the Hive Console.
3. All features of Hive Console (the dashboard) are available for every type of account. Here's the relevant documenation for the two:
  a. [Explorer](https://the-guild.dev/graphql/hive/docs/dashboard/explorer)
  b. [Laboratory](https://the-guild.dev/graphql/hive/docs/dashboard/laboratory)
4. The Hive Gateway CLI (`npx hive-gateway`) already has all performance related features implemented. You're recommended to use either the CLI or the official Docker image from this repo.

**Comment by enisdenjo** [comment]
Oh, @ardatan beat me to it! @vibhor-d11 pick and choose. 😄
